utils/common/src/main/java/brooklyn/util/yaml/Yamls.java

```diff
@@ -66,6 +66,47 @@
         return (T)x;
     }
 
+    /**
+     * Parses the given yaml, and walks the given path to return the referenced object.
+     * 
+     * @see #getAt(Object, List)
+     */
+    public static Object getAt(String yaml, List<String> path) {
+        Iterable<Object> result = new org.yaml.snakeyaml.Yaml().loadAll(yaml);
+        Object current = result.iterator().next();
+        return getAtPreParsed(current, path);
+    }
+    
+    /** 
+     * For pre-parsed yaml, walks the maps/lists to return the given sub-item.
+     * In the given path:
+     * <ul>
+     *   <li>A vanilla string is assumed to be a key into a map.
+     *   <li>A string in the form like "[0]" is assumed to be an index into a list
+     * </ul>
+     * 
+     * Returns {@code null} if that path does not exist. 
+     */
+    @SuppressWarnings("unchecked")
+    public static Object getAtPreParsed(Object current, List<String> path) {
```

Comment:
ugly method name.  if it has already been parsed is there any difference between this and `Jsonya.of(current).at(path).get()` ?

(at least mark `@Beta`)